### ISSUE TITLE
How do I dynamically link libraries in extension?

### ISSUE THREAD

**Issue by rupurt** [duckdb/extension-template#24]
I'm trying to build an ODBC scanner extension and I need to link to `unixodbc`. I can't figure out how to link it in `CMakeLists.txt`. I get an `undefined reference` during the build.

https://github.com/rupurt/odbc-scanner-duckdb-extension/pull/2/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR10

```sh
[ 78%] Linking CXX executable test_sqlite3_api_wrapper
/usr/bin/ld: ../../src/libduckdb.so: undefined reference to `SQLDriverConnect'
collect2: error: ld returned 1 exit status
gmake[3]: *** [tools/sqlite3_api_wrapper/CMakeFiles/test_sqlite3_api_wrapper.dir/build.make:128: tools/sqlite3_api_wrapper/test_sqlite3_api_wrapper] Error 1

```

**Comment by samansmink** [comment]
Hi @rupurt. I don't really know much about nix, but I'm currently working on integrating vcpkg in our extensions builds to do dependency management. I'm expecting to add this to the extension template in the coming weeks. If you want to have a go at it before I add it to the template, you could check out https://github.com/samansmink/aws, an experimental aws extension that should give you enough pointers to set it up yourself

Btw, you know about the ODBC client in main duckdb? https://github.com/duckdb/duckdb/blob/master/tools/odbc/README.md

**Comment by rupurt** [comment]
Thanks @samansmink.

I managed to figure it out in the end. I had to bump the minimum `cmake` version to at least `3.13.0` which uses a new policy that allows you to [link to targets outside the current project](https://github.com/rupurt/odbc-scanner-duckdb-extension/pull/2/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR16).

I can add a new PR with documentation to this in the `README` if you like or can bump the min version in the template.
